src/BSON/Binary.c

```diff
@@ -199,6 +205,96 @@ PHP_METHOD(Binary, getType)
 }
 /* }}} */
 
+/* {{{ proto string Binary::serialize()
+*/
+PHP_METHOD(Binary, serialize)
+{
+	php_phongo_binary_t      *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+
+	intern = Z_BINARY_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+#if PHP_VERSION_ID >= 70000
+	array_init_size(&retval, 2);
+	ADD_ASSOC_STRINGL(&retval, "data", intern->data, intern->data_len);
+	ADD_ASSOC_LONG_EX(&retval, "type", intern->type);
+#else
+	ALLOC_INIT_ZVAL(retval);
+	array_init_size(retval, 2);
+	ADD_ASSOC_STRINGL(retval, "data", intern->data, intern->data_len);
+	ADD_ASSOC_LONG_EX(retval, "type", intern->type);
+#endif
+
+	PHP_VAR_SERIALIZE_INIT(var_hash);
+	php_var_serialize(&buf, &retval, &var_hash TSRMLS_CC);
+	smart_str_0(&buf);
+	PHP_VAR_SERIALIZE_DESTROY(var_hash);
+
+	PHONGO_RETVAL_SMART_STR(buf);
+
+	smart_str_free(&buf);
+	zval_ptr_dtor(&retval);
+}
+/* }}} */
+
+/* {{{ proto string Binary::unserialize(string $serialized)
+*/
+PHP_METHOD(Binary, unserialize)
+{
+	php_phongo_binary_t    *intern;
+	zend_error_handling     error_handling;
+	char                   *serialized;
+	phongo_zpp_char_len     serialized_len;
+#if PHP_VERSION_ID >= 70000
+	zval                    props;
+#else
+	zval                   *props;
+#endif
+	php_unserialize_data_t  var_hash;
+
+	intern = Z_BINARY_OBJ_P(getThis());
+
+	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
+	intern = Z_BINARY_OBJ_P(getThis());
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &serialized, &serialized_len) == FAILURE) {
+		zend_restore_error_handling(&error_handling TSRMLS_CC);
+		return;
+	}
+	zend_restore_error_handling(&error_handling TSRMLS_CC);
+
+#if PHP_VERSION_ID < 70000
+	ALLOC_INIT_ZVAL(props);
+#endif
+	PHP_VAR_UNSERIALIZE_INIT(var_hash);
+	if (!php_var_unserialize(&props, (const unsigned char**) &serialized, (unsigned char *) serialized + serialized_len, &var_hash TSRMLS_CC)) {
+		zval_ptr_dtor(&props);
+		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "%s deserialization failed", ZSTR_VAL(php_phongo_binary_ce->name));
+
+		PHP_VAR_UNSERIALIZE_DESTROY(var_hash);
+		RETURN_FALSE;
```

Comment:
[`Serializable::unserialize()`](http://php.net/manual/en/serializable.unserialize.php) has a void return type, so I think we can simply use `return` here.